src/domain/applications/applicationDetails/ApplicationDetails.tsx

```diff
@@ -0,0 +1,149 @@
+}
+
+const ApplicationDetails: React.SFC<ApplicationDetailsProps> = ({
+  applicationType,
+  createdAt,
+  queue,
+  status,
+  boatType,
+  boatRegistrationNumber,
+  boatWidth,
+  boatLength,
+  boatDraught,
+  boatWeight,
+  boatName,
+  boatModel,
+  harborChoices,
+  accessibilityRequired,
+}) => {
+  const { t } = useTranslation();
+  const notNull = (choice: HarborChoice | null): choice is HarborChoice =>
+    !!choice;
+
+  return (
+    <div className={styles.applicationsDetails}>
+      <Grid colsCount={3}>
+        <Section title={t('applications.applicationDetails.application')}>
+          <LabelValuePair
+            label={t('applications.applicationDetails.applicationType')}
+            value={applicationType}
+          />
+          <LabelValuePair
+            label={t('applications.applicationDetails.receivedDate')}
+            value={createdAt}
+          />
+          <LabelValuePair
+            label={t('applications.applicationDetails.queueNumber')}
+            value={`${queue || ''}`}
+          />
+          <LabelValuePair
+            label={t('applications.applicationDetails.status')}
+            value={status}
+          />
+        </Section>
+        <div>
+          <Section title={t('applications.applicationDetails.boatInfo')}>
+            <LabelValuePair
+              label={t('applications.applicationDetails.boatType')}
+              value={boatType}
+            />
+            <LabelValuePair
+              label={t('applications.applicationDetails.registrationNumber')}
+              value={boatRegistrationNumber}
+            />
+          </Section>
+          <Section>
+            <LabelValuePair
+              label={t('applications.applicationDetails.boatWidth')}
+              value={`${boatWidth}`}
+            />
+            <LabelValuePair
+              label={t('applications.applicationDetails.boatLength')}
+              value={`${boatLength}`}
+            />
+            <LabelValuePair
+              label={t('applications.applicationDetails.boatDepth')}
+              value={`${boatDraught || ''}`}
+            />
+            <LabelValuePair
+              label={t('applications.applicationDetails.boatWeight')}
+              value={`${boatWeight || ''}`}
+            />
+          </Section>
+          <Section>
+            <LabelValuePair
+              label={t('applications.applicationDetails.boatName')}
+              value={boatName}
+            />
+            <LabelValuePair
+              label={t('applications.applicationDetails.boatBrand')}
+              value={boatModel}
+            />
+          </Section>
+        </div>
+        <div>
+          <Section title={t('applications.applicationDetails.selectedPorts')}>
+            <List noBullets>
+              {[...harborChoices]
+                .filter(notNull)
+                .sort((choiceA, choiceB) => choiceA.priority - choiceB.priority)
+                .map(({ harborName, harbor }, i) => (
```

Comment:
This look quite expensive
How about do filter in map -> sort result?

Reply:
Yes, I can combine them in `reduce`, initially I opted for this solution for the sake of readability since `harborChoices` might contain only a handful of items.